src/hotspot/share/opto/parse1.cpp

```diff
@@ -401,19 +401,17 @@ Parse::Parse(JVMState* caller, ciMethod* parse_method, float expected_uses)
   _wrote_stable = false;
   _wrote_fields = false;
   _alloc_with_final = nullptr;
-  _entry_bci = InvocationEntryBci;
-  _tf = nullptr;
   _block = nullptr;
   _first_return = true;
   _replaced_nodes_for_exceptions = false;
   _new_idx = C->unique();
-  debug_only(_block_count = -1);
-  debug_only(_blocks = (Block*)-1);
+  DEBUG_ONLY(_entry_bci = UnknownBci);
+  DEBUG_ONLY(_block_count = -1);
+  DEBUG_ONLY(_blocks = (Block*)-1);
 #ifndef PRODUCT
   if (PrintCompilation || PrintOpto) {
     // Make sure I have an inline tree, so I can print messages about it.
-    JVMState* ilt_caller = is_osr_parse() ? caller->caller() : caller;
-    InlineTree::find_subtree_from_root(C->ilt(), ilt_caller, parse_method);
+    InlineTree::find_subtree_from_root(C->ilt(), caller, parse_method);
```

Comment:
Reading this again, you sure that we don't need `caller->caller()` on `is_osr_parse()` path?

Reply:
first of all, is_osr_parse() was false at line 415 because _entry_bci was assigned to InvocationEntryBci right before.  That's why I use *caller* directly. 

Even we consider to build InlineTree for OSR, I don't think caller->caller() is correct.
I explain this in item 2 here.
https://github.com/openjdk/jdk/pull/16669#issuecomment-1820258714

Reply:
Ah OK, trippy... All good then.